sympy/stats/crv_types.py

```diff
@@ -2195,6 +2196,113 @@ def Normal(name, mean, std):
             MultivariateNormalDistribution, name, mean, std)
     return rv(name, NormalDistribution, (mean, std))
 
+
+#-------------------------------------------------------------------------------
+# Inverse Normal distribution ----------------------------------------------------------
+
+
+class NormalInverseDistribution(SingleContinuousDistribution):
```

Comment:
Do you prefer this name to `InverseNormalDistribution`?

Reply:
I preferred `InverseNormalDistribution` initially but there is Inverse Gamma distribution already declared and it is named `GammaInverseDistribution`, so I have used `NormalInverse`. What do you think about it?

Reply:
OK, I felt `InverseNormal` and `InverseGamma` would have been better names. But it's fine if there are other distributions named along those lines.

Reply:
By the way, it seems Inverse Gaussian and Inverse Normal can mean different things. In literature, it is recognised as Inverse Gaussian Distribution. I can understand you trying to keep the name to have `Normal` and `NormalInverse` for clarity, but the latter is not quite accurate.
In fact, Wolfram Language also uses the names `NormalDistribution` and `InverseGaussianDistribution`.

Reply:
So, should I change this? I was also of the opinion that this should be called Inverse Gaussian distribution, but to keep the name similar to Normal and having read [here](https://stats.stackexchange.com/questions/55962/what-is-the-difference-between-a-normal-and-a-gaussian-distribution) that Normal and Gaussian are synonymous, I went for `NormalInverse`. What do you think about it? Changing the name to `GaussianInverse` is fine with me.

Reply:
> Normal and Gaussian are synonymous, I went for NormalInverse

There are no mentions of this usage for Inverse Gaussian distribution on [Wikipedia](https://en.wikipedia.org/wiki/Inverse_Gaussian_distribution) and [Wolfram Mathworld](http://mathworld.wolfram.com/InverseGaussianDistribution.html).

Quoting from https://www.statisticshowto.datasciencecentral.com/inverse-normal-distribution/:
> The inverse normal distribution is not the same thing as the Inverse Gaussian distribution. The inverse normal distribution refers to the technique of working backwards to find x-values. In other words, you’re finding the inverse. The inverse Gaussian is a two-parameter family of continuous probability distributions.

It seems that is how inverse is perceived generally with normal distribution. Also, see https://www.johndcook.com/blog/normal_cdf_inverse/. `GaussianInverse` seems to be mainstream with the possibility of considering `Wald`. @Upabjojr thoughts?

Reply:
_NormalInverse_ definitely sounds like a bad name, but if you think that's good, we can go for it.

Reply:
@sidhantnagpal thank you for the resource. I will be changing the name to `GaussianInverse`.